open_spiel/games/rbc.cc

```diff
@@ -275,43 +275,46 @@ RbcState::RbcState(std::shared_ptr<const Game> game, int board_size,
                    const std::string& fen)
     : State(game),
       start_board_(*chess::ChessBoard::BoardFromFEN(
-          fen, board_size, /*king_in_check_allowed=*/true)),
+          fen, board_size,
+          /*king_in_check_allowed=*/true,
+          /*allow_pass_move=*/true)),
       current_board_(start_board_),
       phase_(MovePhase::kSensing) {
   SPIEL_CHECK_TRUE(&current_board_);
   repetitions_[current_board_.HashValue()] = 1;
 }
 
 void RbcState::DoApplyAction(Action action) {
+  // Reset common flags.
+  illegal_move_attempted_ = false;
+  move_captured_ = false;
+
   if (phase_ == MovePhase::kSensing) {
     sense_location_[CurrentPlayer()] = action;
     phase_ = MovePhase::kMoving;
-    illegal_move_attempted_ = false;
-    move_captured_ = false;
   } else {
     chess::Move move = ActionToMove(action, Board());
 
     // Handle special cases for RBC.
-    if (action == chess::kPassAction) {
-      move = chess::kPassMove;
-    } else if (Board().IsBreachingMove(move)) {
+    // The RBC's pass move is handled via ChessBoard flag allow_pass_move.
+    if (Board().IsBreachingMove(move)) {
       SPIEL_DCHECK_FALSE(Board().IsMoveLegal(move));
       Board().BreachingMoveToCaptureMove(&move);
       // Transformed move must be legal.
       SPIEL_DCHECK_TRUE(Board().IsMoveLegal(move));
-      // And it must be a capture.
+      // And it must be a capture, since we breached unseen opponent pieces.
       SPIEL_DCHECK_NE(Board().at(move.from).color, Board().at(move.to).color);
+      move_captured_ = true;
     } else if (!Board().IsMoveLegal(move)) {
       illegal_move_attempted_ = true;
-      move = chess::kPassMove;
+      move = chess::kPassMove;  // Illegal move was chosen, so pass instead.
+    } else {
+      // All other moves
+      SPIEL_DCHECK_NE(Board().at(move.from).color, chess::Color::kEmpty);
+      move_captured_ = Board().at(move.from).color != Board().at(move.to).color;
```

Comment:
The problem was here. On a pass move, the move.from and move.to here were both -1, so the index into the array was failing. I'm surprised valgrind did not find that, maybe you need to have a flag for checking array bounds. Anyway I added a special case for the pass move and now it's fine.